modules/mgclient.mjs

```diff
@@ -0,0 +1,1799 @@
+ * @param {All Mg* data types} value.
+ * @return {MgValue}.
+ */
+function errorOrPushToResourceManager(value) {
+    if (value.transferToWasm() == 0) {
+        return null
+    }
+    resourceManager.push(value);
+    return value;
+}
+
+let instance = null;
+
+/**
+ * Initializes mgclient. Users shall call this function before using any of the module exports.
+ * @param  {module} wasmInstance. The instance object retrieved from calling factory().then( instance => {} )
+ * @return {bool}. Returns true on success, false on failure.
+ */
+export function init(wasmInstance) {
+    if (instance != null) {
+        return null;
+    }
+    instance = wasmInstance;
+    return (instance._mg_init() == 0) ? true : false;
+}
+
+/**
+ * Returns the mgclient version.
+ * @return {string} 
+ */
+export function clientVersion() {
+    let wrappedFun = instance.cwrap('mg_client_version', 'string');
+    return wrappedFun();
+}
+
+/**
+ * Releases the internal memory allocated by the wasm mgclient.
+ */
+function releaseAll() {
+    resourceManager.releaseAll();
+}
+
+/**
+ * Finalizes and releases the internal resources. must be called by the users
+ * before the program executes.
+ */
+export function finalize() {
+    resourceManager.releaseAll();
+    instance._mg_finalize();
+}
+
+/**
+ * Returns the total memory resources held. Users shall not use this function.
+ * @return {int}
+ */
+export function resourcesTracked() {
+    return resourceManager.resourcesTracked();
+}
+
+/**
+ * Enum representing the possible data types an MgValue can hold.
+ */
+export class MgValueTypeEnum {
+    static MG_VALUE_TYPE_NULL = new MgValueTypeEnum("MG_VALUE_TYPE_NULL");
+    static MG_VALUE_TYPE_BOOL = new MgValueTypeEnum("MG_VALUE_TYPE_BOOL");
+    static MG_VALUE_TYPE_INTEGER = new MgValueTypeEnum("MG_VALUE_TYPE_INTEGER");
+    static MG_VALUE_TYPE_FLOAT = new MgValueTypeEnum("MG_VALUE_TYPE_FLOAT");
+    static MG_VALUE_TYPE_STRING = new MgValueTypeEnum("MG_VALUE_TYPE_STRING");
+    static MG_VALUE_TYPE_LIST = new MgValueTypeEnum("MG_VALUE_TYPE_LIST");
+    static MG_VALUE_TYPE_MAP = new MgValueTypeEnum("MG_VALUE_TYPE_MAP");
+    static MG_VALUE_TYPE_NODE = new MgValueTypeEnum("MG_VALUE_TYPE_NODE");
+    static MG_VALUE_TYPE_RELATIONSHIP =
+        new MgValueTypeEnum("MG_VALUE_TYPE_RELATIONSHIP");
+    static MG_VALUE_TYPE_UNBOUND_RELATIONSHIP =
+        new MgValueTypeEnum("MG_VALUE_TYPE_UNBOUND_RELATIONSHIP");
+    static MG_VALUE_TYPE_PATH = new MgValueTypeEnum("MG_VALUE_TYPE_PATH");
+    static MG_VALUE_TYPE_DATE = new MgValueTypeEnum("MG_VALUE_TYPE_DATE");
+    static MG_VALUE_TYPE_TIME = new MgValueTypeEnum("MG_VALUE_TYPE_TIME");
+    static MG_VALUE_TYPE_LOCAL_TIME =
+        new MgValueTypeEnum("MG_VALUE_TYPE_LOCAL_TIME");
+    static MG_VALUE_TYPE_DATE_TIME =
+        new MgValueTypeEnum("MG_CALUE_TYPE_DATE_TIME");
+    static MG_VALUE_TYPE_DATE_TIME_ZONE_ID =
+        new MgValueTypeEnum("MG_VALUE_TYPE_DATE_TIME");
```

Comment:
`MG_VALUE_TYPE_DATE_TIME_ZONE_ID` should be used here?

Reply:
We do not support `MG_VALUE_TYPE_DATE_TIME_ZONE_ID` in `memgraph` or `mgclient`. This was for completeness but good catch!